datalake/src/main/java/com/sequenceiq/datalake/service/recovery/RecoveryService.java

```diff
@@ -0,0 +1,53 @@
+package com.sequenceiq.datalake.service.recovery;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.dto.NameOrCrn;
+import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
+import com.sequenceiq.sdx.api.model.SdxRecoveryRequest;
+import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
+
+/**
+ * Defines a service that can be used to recover CB deployed infrastructure.
+ *
+ * Services that can recovery failed infrastructure must be able to <em>validate</em> that a recovery action is appropriate and
+ * be able to <em>trigger</em> the recovery.
+ *
+ * The recovery request should be a value object representing an HTTP response object.
+ *
+ */
+public interface RecoveryService {
+    /**
+     * Starts recovery of the CB infrastructure related to the recovery request.
+     *
+     * This should validate that recovery is appropriate before triggerng the recovery.
```

Comment:
This line is a little confusing since below you say that validation should have already been performed. Are you talking about a different type of validation here?

Also: triggering**.